handlers/secrets.go

```diff
@@ -75,3 +81,86 @@ func makeSecretsArray(c *client.Client, secretNames []string) ([]*swarm.SecretRe
 
 	return values, nil
 }
+
+func secretGetHandler(c *client.Client, w http.ResponseWriter, r *http.Request) {
+	secrets, err := c.SecretList(context.Background(), types.SecretListOptions{})
+
+	if err != nil {
+		w.WriteHeader(http.StatusInternalServerError)
+
+		return
+	}
+
+	var results []requests.Secret
+
+	for _, s := range secrets {
+		results = append(results, requests.Secret{Name: s.Spec.Name})
+	}
+
+	bs, marshalErr := json.Marshal(results)
+
+	if marshalErr != nil {
+		w.WriteHeader(http.StatusInternalServerError)
+
+		return
+	}
+
+	_, writeErr := w.Write(bs)
+
+	if writeErr != nil {
+		w.WriteHeader(http.StatusInternalServerError)
+
+		return
+	}
+
+	w.WriteHeader(http.StatusOK)
+}
+
+func secretPostHandler(c *client.Client, w http.ResponseWriter, r *http.Request) {
+	bs, err := ioutil.ReadAll(r.Body)
+
+	if err != nil {
+		w.WriteHeader(http.StatusBadRequest)
+
+		return
+	}
+
+	var secret requests.Secret
+
+	unmarshalErr := json.Unmarshal(bs, &secret)
+
+	if unmarshalErr != nil {
+		w.WriteHeader(http.StatusBadRequest)
```

Comment:
How about writing the error too?

Reply:
Do we have a standard error response struct yet?

Reply:
Was just comparing to the k8s implementation we have, right now it just logs the errors but does not return the error in the response body. At the minimum we should log each error

Reply:
As a rule we should:

- log the error
- write the error to the caller
- write an error code

This counts for the Kubernetes implementation too @stefanprodan @LucasRoesler

Reply:
I'm on it! :-)

Reply:
Done!